packages/cloud_firestore/ios/Classes/CloudFirestorePlugin.m

```diff
@@ -183,8 +183,7 @@ - (id)readValueOfType:(UInt8)type {
     case DATE_TIME: {
       SInt64 value;
       [self readBytes:&value length:8];
-      NSTimeInterval time = [NSNumber numberWithLong:value].doubleValue / 1000.0;
-      return [NSDate dateWithTimeIntervalSince1970:time];
+      return [NSDate dateWithTimeIntervalSince1970:(value / 1000.0)];
```

Comment:
Just out of curiosity, what is the difference between [NSNumber numberWithLong:value].doubleValue / 1000.0 and value / 1000.0

Reply:
See https://github.com/flutter/flutter/issues/18470

Reply:
My understanding is that one of them is an Objective-C object and the other is an unboxed value.